indexer/feature.cpp

```diff
@@ -27,6 +28,94 @@ void FeatureBase::Deserialize(feature::LoaderBase * pLoader, TBuffer buffer)
   m_header = m_pLoader->GetHeader();
 }
 
+FeatureType FeatureType::FromXML(string const & xml)
+{
+  pugi::xml_document document;
+  document.load(xml.data());
+  return FromXML(document);
+}
+
+FeatureType FeatureType::FromXML(editor::XMLFeature const & xml)
+{
+  FeatureType feature;
+
+  feature.m_center = xml.GetCenter();
+
+  xml.ForEachName([&feature](string const & lang, string const & name)
+                  {
+                    feature.m_params.name.AddString(lang, name);
+                  });
+  feature.m_params.house.Set(xml.GetHouse());
+
+  // TODO(mgsergio):
+  // feature.m_params.ref =
+  // feature.m_params.layer =
+  // feature.m_params.rank =
+
+  // EGeomType
+
+  // for (auto const i : range(feature.GetTypesCount()))
+  //   m_types[i] =
+  // Does the order matters? If so what order should be?
```

Comment:
Only @vng knows that :) It's better to assume that order matters and serialize it too.